src/MomentoTaggedCache.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Momento\Cache;
+
+use Illuminate\Cache\TaggedCache;
+
+class MomentoTaggedCache extends TaggedCache
+{
+
+    public function put($key, $value, $ttl = null)
+    {
+        $tags = $this->tags->getNames();
+        $cacheName = $this->store->getCacheName();
+        foreach ($tags as $tag) {
+            $response = $this->store->setAdd($cacheName, $tag, $key, false, $ttl);
+            if (!$response) {
+                return false;
+            }
+        }
+        $putResponse = $this->store->put($key, $value, $ttl);
+        if (!$putResponse) {
+            return false;
+        }
+        return true;
+    }
+
+    public function get($key, $default = null): mixed
+    {
+        $tags = $this->tags->getNames();
+        $cacheName = $this->store->getCacheName();
+        $value = null;
+        foreach ($tags as $tag) {
+            $keys = $this->store->setFetch($cacheName, $tag);
+            foreach ($keys as $k) {
+                if ($k == $key) {
+                    $value = $this->store->get($key);
```

Comment:
Maybe I'm misunderstanding how tagged cache get should work here...
When ` Cache::tags(['weather', 'seattle'])->get($cityWeatherIdInfo);` is called, only the value for the key tagged with both `weather` and `seattle` needs to be returned?
In other word, if there is a key only tagged with `weather`, we should return `null` as its value?
Or as long as one of the tags are tagged to a key, we can return the value for that key?

But even with my current understanding, I need to update this for loop so that `null` will be assigned when `$k != $key`.

Reply:
oh wow, yeah I think your understanding was closer than mine:

> Cache tags allow you to tag related items in the cache and then flush all cached values that have been assigned a given tag. You may access a tagged cache by passing in an ordered array of tag names. Items stored via tags may not be accessed without also providing the tags that were used to store the value. For example, let's access a tagged cache and put a value into the cache:

So it should indeed only be returning one item, you are correct.  But in that case you can break out of the loop at the point where you find a match.

We might need to think a bit more about the implementation for this.  It will be easier to do when we have a way to write tests.  So I think I will just give a thumbs up on this for now and we may need to re-work it once you are able to write tests.